src/DynamoCoreWpf/ViewModels/Core/DynamoViewModel.cs

```diff
@@ -1771,6 +1854,71 @@ private bool CanShowOpenDialogAndOpenResultCommand(object parameter)
         }
 
+
+        /// <summary>
+        /// Present the open dialog and open the workspace that is selected.
+        /// </summary>
+        /// <param name="parameter"></param>
+        private void ShowInsertDialogAndInsertResult(object parameter)
+        {
+            var currentWorkspaceModelType = Model.CurrentWorkspace.GetType();
+            var fileExtensions = currentWorkspaceModelType == typeof(CustomNodeWorkspaceModel) ? "*.dyf" : "*.dyn";
+
+            DynamoOpenFileDialog _fileDialog = new DynamoOpenFileDialog(this, false)
+            {
+                Filter = string.Format(Resources.FileDialogDynamoDefinitions,
+                         BrandingResourceProvider.ProductName, fileExtensions) + "|" +
+                         string.Format(Resources.FileDialogAllFiles, "*.*"),
+                Title = string.Format("Insert Title (Replace)", BrandingResourceProvider.ProductName)
+                //Title = string.Format(Resources.OpenDynamoDefinitionDialogTitle, BrandingResourceProvider.ProductName)
+            };
+
+            // if you've got the current space path, use it as the inital dir
+            if (!string.IsNullOrEmpty(Model.CurrentWorkspace.FileName))
+            {
+                string path = Model.CurrentWorkspace.FileName;
+                if (File.Exists(path))
+                {
+                    var fi = new FileInfo(Model.CurrentWorkspace.FileName);
+                    _fileDialog.InitialDirectory = fi.DirectoryName;
+                }
+                else
+                {
+                    _fileDialog.InitialDirectory = Environment.GetFolderPath(Environment.SpecialFolder.MyComputer);
+                }
+            }
+            else // use the samples directory, if it exists
+            {
+                Assembly dynamoAssembly = Assembly.GetExecutingAssembly();
+                string location = Path.GetDirectoryName(dynamoAssembly.Location);
+                string UICulture = CultureInfo.CurrentUICulture.Name;
+                string path = Path.Combine(location, "samples", UICulture);
+
+                if (Directory.Exists(path))
+                    _fileDialog.InitialDirectory = path;
+            }
+
+            if (HomeSpace.RunSettings.RunType != RunType.Manual)
+            {
+                MainGuideManager.CreateRealTimeInfoWindow("Example file added to workspace. Run mode changed to Manual.", true);
+                HomeSpace.RunSettings.RunType = RunType.Manual;
+            }
+
+            if (_fileDialog.ShowDialog() == DialogResult.OK)
+            {
+                if (CanOpen(_fileDialog.FileName))
+                {
+                    Insert(new Tuple<string, bool>(_fileDialog.FileName, _fileDialog.RunManualMode));
+                }
+            }
+        }
+
+        private bool CanShowInsertDialogAndInsertResultCommand(object parameter)
+        {
+            // TODO: Fix condition
+            return HomeSpace.RunSettings.RunEnabled;
```

Comment:
Sounds good to me that, we will let the user insert the workspace when RunEnabled is true. Are you thinking of any other conditions? Otherwise, remove the to-do comment.

Reply:
I added the extra check in case we are still on the Start Page, in which case it doesn't feel correct to have the Insert enabled. I think we briefly touch on that during a meeting, let me know if I should revert back.